rust_dev_preview/sitewise/src/lib.rs

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use std::error::Error as StdError;
+
+#[derive(thiserror::Error, Debug)]
+#[error("unhandled error")]
+pub struct Error {
+    #[source]
+    source: Box<dyn StdError + Send + Sync + 'static>,
+}
+
+impl Error {
+    pub fn unhandled(source: impl Into<Box<dyn StdError + Send + Sync + 'static>>) -> Self {
```

Comment:
Would be nice if there were a way to derive these, but they might be specific to the examples. I'll think more on it.